src/metabase/api/user.clj

```diff
@@ -299,17 +301,19 @@
     (api/checkp (not (db/exists? User, :%lower.email (if email (u/lower-case-en email) email), :id [:not= id]))
                 "email" (tru "Email address already associated to another user."))
     (db/transaction
-     ;; only user or self can update user info
+     ;; only superuser or self can update user info
      ;; implicitly prevent group manager from updating users' info
      (when (or (= id api/*current-user-id*)
                api/*is-superuser?*)
        (api/check-500
         (db/update! User id (u/select-keys-when body
-                                                :present (into #{:locale} (when api/*is-superuser?* [:login_attributes]))
-                                                :non-nil (set (concat [:first_name :last_name :email]
+                                                :present (into #{:first_name :last_name :locale} (when api/*is-superuser?* [:login_attributes]))
+                                                :non-nil (set (concat [:email]
                                                                       (when api/*is-superuser?*
                                                                         [:is_superuser]))))))
-       (maybe-update-user-personal-collection-name! user-before-update first_name last_name))
+       (let [first_name (if (and (nil? first_name) (contains? body :first_name)) :explicitly-nil first_name)
+             last_name  (if (and (nil? last_name) (contains? body :last_name)) :explicitly-nil last_name)]
+         (maybe-update-user-personal-collection-name! user-before-update first_name last_name)))
```

Comment:
If the name key exists and it's value is `nil`, then we know that it was explicitly set to nil, and we rely on the `maybe-update-user-personal-collection-name!` implementation to handle the `:explicitly-nil` key.

Reply:
not needed after adjusting the `updated-user-names` function.